### PR TITLE
Display smart annotations in reports [SCi-826]

### PR DESCRIPTION
**Pull request by mlorb**
https://biosistemika.atlassian.net/browse/SCI-826

### CODE REVIEW

File: app/views/reports/elements/_step_checklist_element.html.erb [hunk 1]
```diff
@@ -8,7 +8,10 @@
           <span class="glyphicon glyphicon-list"></span>
       </div>
       <div class="pull-left checklist-name">
-        <em><%=t 'projects.reports.elements.step_checklist.checklist_name', name: checklist.name %></em>
+        <%= auto_link(smart_annotation_parser(simple_format(checklist.name)),
+                      link: :urls,
+                      sanitize: false,
+                      html: { target: '_blank' }).html_safe %>
```
Comment (mz3944) [inline app/views/reports/elements/_step_checklist_element.html.erb]: You didn't use locale 'projects.reports.elements.step_checklist.checklist_name'.

File: app/views/reports/elements/_step_element.html.erb [hunk 2]
```diff
@@ -27,9 +27,10 @@
     <div class="row">
       <div class="col-xs-12 ql-editor">
         <% if strip_tags(step.description).present? %>
-          <%= auto_link(step.description,
-                        link: :urls,
-                        html: { target: '_blank' }).html_safe %>
+          <%= auto_link(smart_annotation_parser(step.description),
```
Comment (mz3944) [inline app/views/reports/elements/_step_element.html.erb]: Forgot the simple_format. Actually, I suggest to embed simple_format call into the smart_annotation_parser, as it should be used everytime.

File: app/views/results/_result_text.html.erb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <div class="ql-editor">
   <%= auto_link(smart_annotation_parser(result.result_text.text),
```
Comment (mz3944) [inline app/views/results/_result_text.html.erb]: Again, no simple_format.

File: app/views/reports/elements/_step_element.html.erb (hunk 2 above)
Comment (Ducz0r) [inline app/views/reports/elements/_step_element.html.erb]: This should be done globally, if at all, and I suggest we add it as a separate task (it's not specifically related to this issue, thus this PR).

**Comment by ZmagoD** [comment]
:banana:

File: app/views/reports/elements/_step_element.html.erb (hunk 2 above)
Comment (mz3944) [inline app/views/reports/elements/_step_element.html.erb]: That's what I'm sayin, man!

Comment (mz3944) [inline app/views/reports/elements/_step_element.html.erb]: Well @mlorb, then just open an issue for this, bitterschön.

**Comment by mlorb** [comment]
ok now? @Ducz0r @mz3944 @ZmagoD :banana:

**Comment by Ducz0r** [comment]
Checking...

**Comment by Ducz0r** [comment]
It works, meeeerge! :+1:

**Pull request merged**